cmd/soroban-rpc/internal/preflight/pool.go

```diff
@@ -127,12 +127,12 @@ type metricsLedgerEntryWrapper struct {
 	ledgerEntriesFetched uint32
 }
 
-func (m *metricsLedgerEntryWrapper) GetLedgerEntry(key xdr.LedgerKey, includeExpired bool) (bool, xdr.LedgerEntry, error) {
+func (m *metricsLedgerEntryWrapper) GetLedgerEntries(includeExpired bool, keys ...xdr.LedgerKey) ([]db.LedgerKeyAndEntry, error) {
 	startTime := time.Now()
-	ok, entry, err := m.LedgerEntryReadTx.GetLedgerEntry(key, includeExpired)
+	entries, err := m.LedgerEntryReadTx.GetLedgerEntries(includeExpired, keys...)
 	atomic.AddUint64(&m.totalDurationMs, uint64(time.Since(startTime).Milliseconds()))
 	atomic.AddUint32(&m.ledgerEntriesFetched, 1)
```

Comment:
hmm. that's an interesting observation @tamirms . I think that in this case we should be updating that to 
```golang
atomic.AddUint32(&m.ledgerEntriesFetched, len(entries))
```
I'm not sure if it should be `len(keys)` though, since that's the attempt count vs success count.
possibly would would want both.

Reply:
Oh, very true

Reply:
I think it's ok add the number of keys since you queried for all of them

Reply:
In the same way before we didn't add 0 if the query didn't find the key